libtenzir/builtins/operators/drop.cpp

```diff
@@ -182,6 +184,12 @@ class drop_operator2 final : public crtp_operator<drop_operator2> {
         auto resolved = resolve(sel, slice.schema());
         std::move(resolved).match(
           [&](offset off) {
+            if (off.empty()) {
+              diagnostic::warning("cannot drop `this`")
+                .primary(sel)
+                .emit(ctrl.diagnostics());
+              return;
+            }
```

Comment:
You can detect this statically after argument parsing (also making this an error instead).

Reply:
I pushed a change for this.